### DISCUSSION TITLE
[Task]: MlFlow support

### DISCUSSION THREAD

**Discussion by Ravio1i** [open-edge-platform/anomalib#1206] [Feature Requests] [answered]
### What is the motivation for this task?

We are using MlFlow. Is it planned to support sending logs to mlflow?

### Describe the solution you'd like

```yaml
  logger: mlflow
```

**Comment by samet-akcay** [comment]
@Ravio1i , to do this, we just need to implement the `AnomalibMLFlow` logger class that is similar to following logger implementations:
- [AnomalibWandbLogger](https://github.com/openvinotoolkit/anomalib/blob/main/src/anomalib/utils/loggers/wandb.py)
- [AnomalibTensorBoardLogger](https://github.com/openvinotoolkit/anomalib/blob/main/src/anomalib/utils/loggers/tensorboard.py)
- [AnomalibCometLogger](https://github.com/openvinotoolkit/anomalib/blob/main/src/anomalib/utils/loggers/comet.py)

It would inherit Anomalib's `ImageLoggerBase` and Lightning's `Mlflow` logger, and implement `add_image(self, image: np.ndarray | Figure, name: str | None = None, **kwargs: Any)` method.

```python
from pytorch_lightning.loggers import Mlflow
from anomalib.utils.loggers.base import ImageLoggerBase


class AnomalibMlflow(ImageLoggerBase, Mlflow):
    
    def __init__():
        ...

    @rank_zero_only
    def add_image(self, image: np.ndarray | Figure, name: str | None = None, **kwargs: Any):
        ...
```

**Reply by ninafiona** [reply]
Can you create a merge request? I'm also using it. :pray:

**Reply by ggiret-thinkdeep** [reply]
Hello
Any news about this ? I'm interested too ! :)

**Reply by SilverStarCoder** [reply]
@samet-akcay  we are also using MlFlow, this would be a great addition 🙏

**Reply by samet-akcay** [reply]
yeah, we would like to add MLFlow support

**Comment by samet-akcay** [comment, marked as answer]
Closing this as it is being addressed here https://github.com/openvinotoolkit/anomalib/issues/1759
